Cloudinary/Actions/BaseResult.cs

```diff
@@ -120,11 +120,13 @@ public abstract class BaseResult
         /// </summary>
         /// <param name="response">HTTP response</param>
         /// <returns>New instance of this class</returns>
-        internal static T Parse<T>(HttpWebResponse response) where T : BaseResult, new()
+        internal static T Parse<T>(Object responseObj) where T : BaseResult, new()
         {
-            if (response == null)
+            if (responseObj == null)
                 throw new ArgumentNullException("response");
 
+            HttpWebResponse response = (HttpWebResponse)responseObj;

```

Comment:
Yes, its implementation is less type-safety.  Unfortunately, it's unavoidable in the unification of this kind but the situation you described cannot arise at this point.
I do agree with you though that we can move this line before "if". It can only do good.